plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueConverter.java

```diff
@@ -176,7 +181,12 @@ else if (isIcebergTable(tableParameters)) {
         }
         else {
             boolean isCsv = sd.serdeInfo() != null && HiveStorageFormat.CSV.getSerde().equals(sd.serdeInfo().serializationLibrary());
-            dataColumns = fromGlueColumns(sd.columns(), ColumnType.DATA, isCsv);
+            if (isDeltaLakeTable(tableParameters) && ignoreDeltaLakeColumns) {
```

Comment:
Similarly to Iceberg handling, consider placing this in a block 

```
else if (isDeltaLakeTable(tableParameters)) {
....
}
```

The `partitionColumns` are not stored AFAIK in the metastore.

Check also whether the same storage as for the Iceberg tables makes sense in this case.
```
storage = FAKE_PARQUET_STORAGE;
```

I'm inclined to advocate for setting always for Iceberg & Delta Lake tables:

```
dataColumns = ImmutableList.of(FAKE_COLUMN);
```

We don't use nowhere this parsed information. The downside of trying to parse this information is that sometimes we may get burned (as in the situation you're trying to fix).

Reply:
> Similarly to Iceberg handling, consider placing this in a block

Why? Can you add a code suggestion?

> Check also whether the same storage as for the Iceberg tables makes sense in this case.

Isn't this out of the scope of this PR?

> I'm inclined to advocate for setting always for Iceberg & Delta Lake tables

I'd agree, but I don't feel confident expanding the scope of this for Iceberg, as I can't write a new test for Iceberg, and I don't know if the existing tests cover that.